org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavadocConverter.java

```diff
@@ -49,22 +51,47 @@
 import com.sun.tools.javac.util.JCDiagnostic;
 
 class JavadocConverter {
-	
+
 	private final AST ast;
 	private final JavacConverter javacConverter;
 	private final DCDocComment docComment;
 	private final int initialOffset;
 	private final int endOffset;
-	
+
 	final private Set<JCDiagnostic> diagnostics = new HashSet<>();
 
+	private static Field UNICODE_READER_CLASS_OFFSET_FIELD = null;
+	static {
+		try {
+			Class<UnicodeReader> unicodeReaderClass = (Class<UnicodeReader>) Class.forName("com.sun.tools.javac.parser.UnicodeReader");
+			UNICODE_READER_CLASS_OFFSET_FIELD = unicodeReaderClass.getDeclaredField("offset");
+			UNICODE_READER_CLASS_OFFSET_FIELD.setAccessible(true);
+		} catch (Exception e) {
+			// do nothing, leave null
+		}
+	}
+
 	JavadocConverter(JavacConverter javacConverter, DCDocComment docComment) {
 		this.javacConverter = javacConverter;
 		this.ast = javacConverter.ast;
 		this.docComment = docComment;
-		this.initialOffset = this.javacConverter.rawText.substring(0, docComment.getSourcePosition(0)).lastIndexOf("/**");
-		int offsetEnd = this.docComment.getSourcePosition(this.docComment.getEndPosition());
-		this.endOffset = offsetEnd + this.javacConverter.rawText.substring(offsetEnd).indexOf("*/") + "*/".length();
+
+		int startPos = -1;
+		if (UNICODE_READER_CLASS_OFFSET_FIELD != null) {
+			try {
+				startPos = UNICODE_READER_CLASS_OFFSET_FIELD.getInt(docComment.comment);
+			} catch (Exception e) {
+				ILog.get().warn("could not reflexivly access doc comment offset");
+			}
+		} else {
+			startPos = docComment.getSourcePosition(0) >= 0 ? docComment.getSourcePosition(0) : docComment.comment.getSourcePos(0);
+		}
+
+		if (startPos < 0) {
+			throw new IllegalArgumentException("Doc comment has no start position");
+		}
+		this.initialOffset = startPos;
+		this.endOffset = startPos + this.javacConverter.rawText.substring(startPos).indexOf("*/") + "*/".length();
```

Comment:
Something that we don't handle properly is that either of these could be written out as unicode and the comment would still be valid eg. the document contains the literal content `\u002A\u002F`